lib/DefinePlugin.js

```diff
@@ -42,6 +42,10 @@ class RuntimeValue {
 }
 
 const stringifyObj = (obj, parser) => {
+	if (Array.isArray(obj)) {
+		return "[" + obj.map(code => toCode(code, parser)).join(",") + "]";
```

Comment:
This can have problems it's at the start of a line and previous line doesn't end with a semicolon.

``` js
x()
ARRAY.forEach(x => console.log(x));
```

would be replaced with

``` js
x()
[1,2,3].forEach(x => console.log(x));
```

wich is equal to

``` js
x()[1,2,3].forEach(x => console.log(x));
```

You could make it replace with `Object([1,2,3])` to be safe.

Reply:
Now, I understand why there is `Object({})` instead of `{}` for the object case!